app/services/base_result.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+class BaseResult
+  def self.[](*attributes)
+    Class.new(BaseResult) { attr_accessor(*attributes) }
```

Comment:
Considered using `Struct` for this vs just class? 🤔

Reply:
Struct does not offers setters, so it would make it harder to use as it would require us to pass the result attributes when creating the instance of the result.